cpp/src/arrow/array/array_test.cc

```diff
@@ -78,20 +79,58 @@ class TestArray : public ::testing::Test {
   MemoryPool* pool_;
 };
 
+template <class ScalarType>
+std::shared_ptr<ScalarType> MakeScalar(typename ScalarType::ValueType value) {
+  return std::make_shared<ScalarType>(value);
+}
```

Comment:
How is this different than `MakeScalar` in `scalar.h`?  If it is different, can we add a comment because I know I'll ask the question again :laughing:

Reply:
I was having trouble getting it to work, but now I've realized I should have just passed `int32_t` instead of `Int32Type` to the template. :)

So I'm removing it now in favor of scalar.h's version.